src/screens/ProfileScreen/elements/RenderItem.js

```diff
@@ -81,24 +81,32 @@ const Item = ({
 
   React.useEffect(() => {
     const validationStatusVote = () => {
-      if (item.reaction_counts) {
-        if (item.latest_reactions.upvotes) {
-          const upvote = item.latest_reactions.upvotes.find((vote) => vote.user_id === selfUserId);
-          if (upvote) {
-            setVoteStatus('upvote');
-            setStatusUpvote(true);
-          }
-        } else if (item.latest_reactions.downvotes) {
-          const downvotes = item.latest_reactions.downvotes.find(
-            (vote) => vote.user_id === selfUserId
-          );
-          if (downvotes) {
-            setVoteStatus('downvote');
-            setStatusDowvote(true);
-          }
-        } else {
-          setVoteStatus('none');
-        }
+      const findUpvote =
+        item &&
+        item.own_reactions &&
+        item.own_reactions.upvotes &&
+        Array.isArray(item.own_reactions.upvotes) &&
+        item.own_reactions.upvotes.find(
+          (reaction) => reaction.user_id === profile.myProfile.user_id
+        );
+      const findDownvote =
+        item &&
+        item.own_reactions &&
+        item.own_reactions.downvotes &&
+        Array.isArray(item.own_reactions.downvotes) &&
+        item.own_reactions.downvotes.find(
+          (reaction) => reaction.user_id === profile.myProfile.user_id
+        );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code for finding upvotes and downvotes is quite repetitive. It would be better to create a helper function that can be reused for both upvotes and downvotes. This will make the code more DRY (Don't Repeat Yourself) and easier to maintain.
```diff
-      const findUpvote =
-        item &&
-        item.own_reactions &&
-        item.own_reactions.upvotes &&
-        Array.isArray(item.own_reactions.upvotes) &&
-        item.own_reactions.upvotes.find(
-          (reaction) => reaction.user_id === profile.myProfile.user_id
-        );
-      const findDownvote =
-        item &&
-        item.own_reactions &&
-        item.own_reactions.downvotes &&
-        Array.isArray(item.own_reactions.downvotes) &&
-        item.own_reactions.downvotes.find(
-          (reaction) => reaction.user_id === profile.myProfile.user_id
-        );

+      const findReaction = (type) =>
+        item?.own_reactions?.[type]?.find(
+          (reaction) => reaction.user_id === profile.myProfile.user_id
+        );
+
+      const findUpvote = findReaction('upvotes');
+      const findDownvote = findReaction('downvotes');
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->